### DISCUSSION TITLE
How do I make the osg-satellites example automatically start the emulation in omnet++6.0.1

### DISCUSSION THREAD

**Discussion by tellbin** [omnetpp/omnetpp#1020] [Q&A] [answered]
Hello, my main development language is java, and now I want to call the osg-satellites example through the command line and perform simulation and then process the simulation result. According to the document, I tried to compile the case into exe and call it through mingwenv. Because this case requires Qtenv, I can start the simulation window but can't start the simulation automatically. I have to click to run it in the simulation window. Can you guide me?

![动画6](https://github.com/omnetpp/omnetpp/assets/9007576/4342e3a6-b8af-4c95-a054-9923299efdf2)

**Comment by tellbin** [comment]
I just want to drive the simulation through the command line and automatically run the specified number of times or a certain amount of time and then end

**Reply by rhornig** [reply]
So why are you running with Qtenv which was specifically designed to be interactive. To run a simulation in the background, you should use Cmdenv.

BUT: The osg-satellites example was created to demonstrate the 3D visualization capabilities. It actually does NOT simulate anything, so running it in Cmdenv makes no sense either.

**Reply by tellbin** [reply]
Thanks Reply.It is just to show that it can be simulated automatically, otherwise it is not easy for customers to understand only by looking at the command line. The purpose is to show that omnet can be integrated. If there is no other way, I can only use Cmdenv

**Reply by rhornig** [reply, marked as answer]
There is no functionality in Qtenv that allows it to be remote controlled by other processes. You can either run Cmdenv by providing parameters to control the length of run/number of runs. Or use Qtenv to run a single run interactively. Simulation in Qtenv cannot be started by a different process.

**Reply by tellbin** [reply]
ok, thanks for the reply
